crypto/fipsmodule/ml_kem/ml_kem_ref/kem.c

```diff
@@ -112,13 +196,17 @@ int crypto_kem_enc_derand(ml_kem_params *params,
 *              - const uint8_t *pk: pointer to input public key
 *                (an already allocated array of KYBER_PUBLICKEYBYTES bytes)
 *
-* Returns 0 (success)
+* Returns 0 (success), or 1 when the encapsulation key check fails.
 **************************************************/
 int crypto_kem_enc(ml_kem_params *params,
                    uint8_t *ct,
                    uint8_t *ss,
                    const uint8_t *pk)
 {
+  if (encapsulation_key_modulus_check(params, pk) != 0) {
+    return 1;
```

Comment:
I looked into this -- the wording in that comment is a bit weird. It says an "invalid public key" but what it means is a public key that doesn't match the corresponding secret key. So I updated the comment. There was another issue with the test code that I fixed as well. The code was "invalidating" the key by negating its first byte. This can make the key actually invalid in the sense of FIPS 203 and with the change in this PR such key would be rejected, while the intention of the test is not to test real invalid keys but valid public keys that are not matching the secret key and thus would fail the final handshake.